File: boj/Silver IV/BOJ_10828/BOJ_10828.cpp
#include <iostream>
#include <stack>

using namespace std;

int n, num;
string str;
stack<int> st;

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n;
    while (n--) {
        // 공백까지 입력받기 위해서
        cin >> str;
        if (str == "pop") {
            // 비어있다면 -1 출력
            if (st.empty()) {
                cout << "-1" << '\n';
            }
            // 비어있지 않다면 st.top()을 출력 후 삭제
            else {
                cout << st.top() << '\n';
                st.pop();
            }
        }
        else if (str == "size") {
            cout << st.size() << '\n';
        }
        else if (str == "empty") {
            if (st.empty()) {
                cout << '1' << '\n';
            }
            else {
                cout << '0' << '\n';
            }
        }
        else if (str == "top") {
            if (st.empty()) {
                cout << "-1" << '\n';
            }
            else {
                cout << st.top() << '\n';
            }
        }
        // str == "push"
        else {
            // X
            cin >> num;
            st.push(num);
        }
    }

    return 0;
}

File: Silver IV/BOJ_10828/BOJ_10828.py
import sys
from collections import deque

if __name__ == "__main__":
    stack = deque()

    n = int(sys.stdin.readline())
    info = [sys.stdin.readline().strip() for _ in range(n)]

    # print(info)

    for row in info:
        if row == "top":
            if stack:
                value = stack.pop()
                print(value)
                stack.append(value)
            else:
                print(-1)
        elif row == "pop":
            if stack:
                print(stack.pop())
            else:
                print(-1)
        elif row == "size":
            print(len(stack))
        elif row[0:4] == "push":
            stack.append(int(row[5:]))
        elif row == "empty":
            if stack:
                print(0)
            else:
                print(1)
